src/datadog/tracer.cpp

```diff
@@ -83,15 +83,27 @@ Tracer::Tracer(const FinalizedTracerConfig& config,
   }
 }
 
+namespace {
+template <typename T>
+std::vector<std::string> to_string_vec(const std::vector<T>& vec) {
+  std::vector<std::string> result{vec.size()};
+  result.reserve(vec.size());
+  std::transform(vec.begin(), vec.end(), result.begin(), [](const T& item) {
+    return std::string{to_string_view(item)};
+  });
+  return result;
+}
+}  // namespace
+
 std::string Tracer::config() const {
   // clang-format off
   auto config = nlohmann::json::object({
     {"version", tracer_version_string},
     {"runtime_id", runtime_id_.string()},
-    {"collector", collector_->config()},
+    {"collector", nlohmann::json::parse(collector_->config())},
     {"span_sampler", span_sampler_->config_json()},
-    {"injection_styles", join_propagation_styles(injection_styles_)},
-    {"extraction_styles", join_propagation_styles(extraction_styles_)},
+    {"injection_styles", to_string_vec(injection_styles_)},
```

Comment:
Can you confirm the change?
| Before | After |
| ------ | ----- |
|"tracecontext,datadog"|["tracecontext", "datadog"]| 

If so, thank you—this is indeed an improvement. However, the best approach would be to leverage the [JSON library](https://github.com/nlohmann/json?tab=readme-ov-file#basic-usage).

The code should look something like:
```cpp
void to_json(nlohmann::json& j, const PropagationStyle& style) {
   j = to_string_view(style);
}
```

And the usage:
```cpp
...
{"injection_styles", injection_styles_},
...
```

Reply:
Yes, that's the change. This also fixes the test `TestConfiguration.test_in_http` in nginx. With the changes made in this PR it now looks normal:

```json
{
  "collector": {
    "config": {
      "event_scheduler": {
        "type": "datadog::nginx::NgxEventScheduler"
      },
      "flush_interval_milliseconds": 2000,
      "http_client": {
        "type": "datadog::tracing::Curl"
      },
      "remote_configuration_url": "http://bogus:1234/v0.7/config",
      "request_timeout_milliseconds": 2000,
      "shutdown_timeout_milliseconds": 2000,
      "telemetry_url": "http://bogus:1234/telemetry/proxy/api/v2/apmtelemetry",
      "traces_url": "http://bogus:1234/v0.4/traces"
    },
    "type": "datadog::tracing::DatadogAgent"
  },
  "defaults": {
    "environment": "fooment",
    "service": "foosvc",
    "service_type": "web"
  },
  "environment_variables": {
    "DD_REMOTE_CONFIG_POLL_INTERVAL_SECONDS": "1"
  },
  "extraction_styles": [
    "B3",
    "Datadog"
  ],
  "injection_styles": [
    "B3",
    "Datadog"
  ],
  "report_traces": true,
  "runtime_id": "31998223-6dc3-4fd6-b4c2-cd1ded711143",
  "span_sampler": {
    "rules": []
  },
  "tags_header_size": 512,
  "trace_sampler": {
    "max_per_second": 200,
    "rules": []
  },
  "version": "[dd-trace-cpp version v0.2.2]"
}
```

Reply:
Indeed. Thanks, I appreciate.